client/network/src/lib.rs

```diff
@@ -294,6 +296,12 @@ pub trait NetworkStateInfo {
 
 	/// Returns the local Peer ID.
 	fn local_peer_id(&self) -> PeerId;
+
+	/// Returns the local Peer PublicKey.
+	fn local_public_key(&self) -> PublicKey;
+
+	/// Returns the peerset
+	fn peerset(&self) -> PeersetHandle;
```

Comment:
I think exposing the peerset in a trait "for providing information about the local network state" (see trait doc comment) is breaking the abstraction. Why not go through the `NetworkService` directly?

In addition, instead of exposing the entire peerset to the outside world, I would suggest keeping it an implementation detail of `client/network`, and only exposing methods on `NetworkService` like `NetworkService::set_priority_group`.

@tomaka probably has more input in this regard.

Reply:
Yes, the peerset is purely an implementation detail. Please don't expose it in the API.

Reply:
Refactored to use Arc<NetworkService>, but leave the `NetworkStateInfo` unchanged since it involves traits used in authority-discovery and I'm not sure the risk to change it. Looks a bit redundant by keeping both in offchain worker though.

Reply:
Cleared the `NetworkStateInfo` use in offchain worker.

Reply:
> Looks a bit redundant by keeping both in offchain worker though.

The idea is to build small coherent abstractions, allowing small components to only depend on subsets of a larger interface.